macrosynergy/panel/basket.py

```diff
@@ -750,12 +751,13 @@ def return_basket(self, basket_names: Union[str, List[str]] = None):
             except KeyError as e:
                 print(f"Basket not found - call make_basket() method first: {e}.")
             else:
-                dfw_retcry = self.column_split(dfw_retcry)
-                dfw_retcry = dfw_retcry.sort_values(["xcat", "real_date"])
-                ret_baskets.append(dfw_retcry)
+                qdf = QuantamentalDataFrame.from_wide(dfw_retcry)
+                ret_baskets.append(qdf)
 
-        return_df = pd.concat(ret_baskets)
-        return return_df.reset_index(drop=True)
+        return_df = QuantamentalDataFrame.from_qdf_list(ret_baskets)
```

Comment:
Flagging this one to discuss later. Wondering if we need both this and the line below.